src/osd/OSDMap.cc

```diff
@@ -2849,8 +2849,34 @@ void OSDMap::print(ostream& out) const
 class OSDTreePlainDumper : public CrushTreeDumper::Dumper<TextTable> {
 public:
   typedef CrushTreeDumper::Dumper<TextTable> Parent;
-  OSDTreePlainDumper(const CrushWrapper *crush, const OSDMap *osdmap_)
-    : Parent(crush), osdmap(osdmap_) {}
+
+  bool down = false;
+  bool up = false;
+  bool in = false;
+  bool out = false;
```

Comment:
this was less code that defining bitfields; not worth optimizing memory usage for a one-off thing like this.